src/System.Composition.Hosting/src/System/Composition/Hosting/Core/ExportDescriptorPromise.cs

```diff
@@ -83,9 +83,9 @@ public ExportDescriptor GetDescriptor()
             _creating = true;
             try
             {
-                var reply = _descriptor.Value;
-                Assumes.IsTrue(reply != null, "Export descriptor fulfillment function returned null.");
-                return reply;
+                ExportDescriptor relay = _descriptor.Value;
+                Requires.NotNull(relay, "getDescriptor");
```

Comment:
That's fair. The only way this would break if people would catch a generic exception and the check the type name of the exception -- I think the number of people who do that is extremely small -- we should be good with the new exception.